```
go build  -gcflags=-trimpath=/build/go/src -buildmode=pie  -o bin/containerd -ldflags '-X github.com/containerd/containerd/version.Version=v1.7.16 -X github.com/containerd/containerd/version.Revision=v1.7.16 -X github.com/containerd/containerd/version.Package=github.com/containerd/containerd -s -w ' -tags "urfave_cli_no_docs"  ./cmd/containerd
# gvisor.dev/gvisor/runsc/container
runsc/container/container.go:1953:18: syntax error: unexpected /, expected expression
error: builder for '/nix/store/yk70p6l70yslfjrxih68sqkr0q4qps89-gvisor-20240401.0.drv' failed with exit code 1;
       last 10 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/l2r5caf44ccj24csdsr2lp1m9lq8hvqb-source
       > source root is source
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > Running phase: buildPhase
       > Building subPackage ./runsc
       > # gvisor.dev/gvisor/runsc/container
       > runsc/container/container.go:1953:18: syntax error: unexpected /, expected expression
```

--- a/pkgs/by-name/gv/gvisor/package.nix
+++ b/pkgs/by-name/gv/gvisor/package.nix
@@ -6,6 +6,7 @@
 , iptables
 , makeWrapper
 , procps
+, glibc
 }:
 
 buildGoModule {
@@ -22,6 +23,12 @@ buildGoModule {
     rev = "9d995324d058812a5476f8c06b20167012511e9c";
     hash = "sha256-idgUEbYAfnm/HphVs12Sj1FwG+jmL2BBr0PJnG9BC3A=";
   };
+
+    # Replace the placeholder with the actual path to ldconfig
+  postPatch = ''
+    substituteInPlace runsc/container/container.go \
+      --replace-fail '"/sbin/ldconfig"' '"${glibc}/bin/ldconfig"'
+  '';
 
   vendorHash = "sha256-jbMXeNXzvjfJcIfHjvf8I3ePjm6KFTXJ94ia4T2hUs4=";
 

